src/Stratis.Bitcoin/Builder/FullNodeBuilder.cs

```diff
@@ -157,6 +157,8 @@ public IFullNode Build()
                 throw new InvalidOperationException("full node already built");
             this.fullNodeBuilt = true;
 
+            this.NodeSettings?.LoadConfiguration();
```

Comment:
It doesn't appear that this.NodeSettings can ever be null here? If you look at the constructors it seems we always have an instance no?

Reply:
There are some test cases that are leaving the NodeSettings null.

Reply:
Some test cases exercise this scenario.

Reply:
Is this still valid here? Seeing as LoadArguments is now private? I assume you can't load configuration without load arguments now?

Reply:
Yes it is still valid. The LoadArguments call happens in the NodeSettings constructor.

Reply:
Yes but what I'm saying is how can you load the config without the args? Isn't it LoadArguments > LoadConfiguration? Pardon, if im missing something here ;)

Reply:
Your question is ambiguous. Either you could have meant:
- You want to know of a way by which you can load the config without loading the arguments.
- You want to know how I am ensuring that the arguments are loaded before the configuration is loaded.
In the first case you can simply omit passing the arguments in the constructor. In the second case it is important to node that you can never call LoadConfiguration before calling LoadArguments due to the latter being called in the constructor.
If this does not answer your question please provide an example.